libsolidity/codegen/ir/Common.h

```diff
@@ -19,14 +19,38 @@
 
 #include <libsolidity/ast/AST.h>
 
+#include <algorithm>
 #include <string>
 
 namespace solidity::frontend
 {
 
+/**
+ * Structure that describes arity and co-arity of a function, i.e. the number of its inputs and outputs.
+ */
+struct Arity
+{
+	size_t in;  /// Number of input parameters
+	size_t out; /// Number of output parameters
+
+	bool operator==(Arity const& _other) const { return in == _other.in && out == _other.out; }
+	bool operator!=(Arity const& _other) const { return !(*this == _other); }
+};
+
 std::string buildFunctionName(FunctionDefinition const& _function);
 std::string buildFunctionName(VariableDeclaration const& _varDecl);
 std::string buildCreationObjectName(ContractDefinition const& _contract);
 std::string buildRuntimeObjectName(ContractDefinition const& _contract);
 
 }
+
+// Overloading std::less() makes it possible to use Arity as a map key. We could define operator<
+// instead but that would be confusing since e.g. Arity{2, 2} would be greater than Arity{1, 10}.
```

Comment:
I would not say it is very confusing...

Reply:
Maybe confusing is not a good word. It's ambiguous. I can think of at least two versions of `<` that make sense:
1. `in1 + out1 < in2 + out2`
2. `in1 < out1 ||  in1 == out1 && in2 < out2`

And in my opinion it's actually (1) that's a more natural choice for this struct but it's not the one I need here.

Reply:
I'm fine with less, but would not mind the direct operator either.

Reply:
I'll leave it at `less` then.